src/api/utils/api_utils.py

```diff
@@ -98,3 +108,283 @@ def create_media_file(file, name):
     media = Media.objects.create(name=name, file=file)
     media.save()
     return media
+
+
+class DonationPageSettings:
+    pass
+
+# -------------------------- Menu Utils --------------------------
+def prependPrefixToLinks(menu_item, prefix):
+    if not menu_item:
+        return None
+    if "link" in menu_item:
+        menu_item["link"] = "/" + prefix + menu_item["link"]
+    if "children" in menu_item:
+        for child in menu_item["children"]:
+            prependPrefixToLinks(child, prefix)
+    return menu_item
+
+def modify_menu_items_if_published(menu_items, page_settings, prefix):
+    if not menu_items or not page_settings or not prefix:
+        return []
+    
+    main_menu = []
+    
+    for item in menu_items:
+        if not item.get("children"):
+            name = item.get("link", "").strip("/")
+            if name in page_settings and not page_settings[name]:
+                main_menu.remove(item)
+        else:
+            for child in item["children"]:
+                name = child.get("link", "").strip("/")
+                if name in page_settings and not page_settings[name]:
+                    item["children"].remove(child)
+        
+    for item in menu_items:
+        f = prependPrefixToLinks(item,prefix)
+        main_menu.append(f)
+
+    return main_menu
+
+def get_viable_menu_items(community):
+
```

Comment:
why isn't DonatePageSettings included in this?

Reply:
It's not currently in use, but I've added it in case we choose to implement it in the future.

Reply:
It is in use, but it is DonatePageSettings, not DonationPageSettings